### PR TITLE
Continuation of "Asciidocj integration"

### PR DESCRIPTION
**Pull request by gillesB**
According to https://github.com/asciidocfx/AsciidocFX/pull/345 the requirements for an Asciidocj integration are:

- [ ] Migration from asciidoctor.js extensions to asciidoctorj extensions
    - [x] chart plugin
    - [ ] math plugin
    - [x] plantuml plugin (but `uml` must be replaced with `plantuml`)
    - [x] ditaa plugin (via `plantuml` block)
    - [ ] tree plugin
- [ ] supporting reveal.js with asciidoctorj
- [x] pdf plugin customization in UI
- [x] switching option from asciidoctorj-pdf to fop by user
- [ ] asciidoctorj should be used in preview and for other backends, like html,
 epub
    - [x] preview
    - [ ] html
    - [ ] docbook
    - [ ] epub
- [x] Outline feature

The checkboxes mark what is implemented in this PR.
As this is my first contribution to this project I did not want to implement everything at once, but ask if I am on the right track before continuing.

As I wanted to reuse as much code as possible, I had to extend `AsciidoctorConfigBase` a bit.
This change is needed because `PdfConfigBean` needs more properties than `AsciidoctorConfigBase` provides.

### CODE REVIEW

**Comment by rahmanusta** [comment]
Hi @gillesB , thanks for the PR. It looks a very nice step to move on AsciidoctorJ. I checked the code out and PDF conversion with AsciidoctorJ has worked well. Keep going the awesome work!

File: src/main/java/com/kodedu/config/EditorConfigBean.java [hunk 1]
```diff
@@ -778,7 +778,7 @@ public JsonObject getJSON() {
                 .add("previewScreenY", getPreviewScreenY())
                 .add("previewScreenWidth", getPreviewScreenWidth())
                 .add("previewScreenHeight", getPreviewScreenHeight())
-                .add("foldStyle", getFoldStyle().name())
+                .add("sStyle", getFoldStyle().name())
```
Comment (rahmanusta) [inline src/main/java/com/kodedu/config/EditorConfigBean.java:781]: @gillesB  Was this an intended change?

Comment (gillesB) [reply src/main/java/com/kodedu/config/EditorConfigBean.java:781]: Sorry, this must have been an unintentional change. I see no reason, why I changed it. Anyway I will revert it in the next commit.

**Comment by rahmanusta** [comment]
@gillesB Reviewing the last commit, it looks perfect!

**Comment by rahmanusta** [comment]
Hi @gillesB  merged changes to asciidoctorj_backend branch, and additionally implemented the tree block processor

**Comment by gillesB** [comment]
Hi @rahmanusta,
thank you for the information. When I continue the work (I do not know when my family allows it) I will base the work on that branch.

**Pull request merged**